core/src/main/java/org/apache/druid/math/expr/ExprMacroTable.java

```diff
@@ -97,9 +99,13 @@ public Expr get(final String functionName, final List<Expr> args)
   {
     protected final Expr arg;
 
+    // Use Supplier to memoize values as ExpressionSelectors#makeExprEvalSelector() can make repeated calls for them
```

Comment:
In the future, please make comments to fields Javadoc comments (with `{@link}` refs when appropriate).